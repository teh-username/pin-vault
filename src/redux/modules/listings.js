import { combineReducers } from 'redux';

export const ADD_ENTRY = 'modules/listings/ADD_ENTRY';
export const MODIFY_ENTRY = 'modules/listings/MODIFY_ENTRY';
export const DELETE_ENTRY = 'modules/listings/DELETE_ENTRY';

/*
  Sample state:
  {
    entries: {
      2cf05d: {
        name: 'Bank 1',
        code: '1234',
      },
      5d94db: {
        name: 'Bank 2',
        code: '5678',
      },
    },
  }
*/

const initialState = {
  entries: {},
};

export const entries = (state = initialState.entries, action) => {
  switch (action.type) {
    case ADD_ENTRY:
      return {
        ...state,
        [action.id]: {
          code: action.code,
          name: action.name,
        },
      };
    case MODIFY_ENTRY:
      return {
        ...state,
        [action.id]: {
          ...state[action.id],
          code: action.code,
        },
      };
    case DELETE_ENTRY:
      return Object.entries(state)
        .filter(([id, entryDetails]) => action.id !== id)
        .reduce((previousState, [id, entryDetails]) => {
          return {
            ...previousState,
            [id]: entryDetails,
          };
        }, {});
    default:
      return state;
  }
};

export default combineReducers({
  entries,
});

export const addEntry = (id, name, code) => ({
  type: ADD_ENTRY,
  id,
  name,
  code,
});

export const modifyEntry = (id, code) => ({
  type: MODIFY_ENTRY,
  id,
  code,
});

export const deleteEntry = id => ({
  type: DELETE_ENTRY,
  id,
});

export const getEntries = ({ listings: { entries } }) =>
  Object.entries(entries);
export const getEntryDetails = ({ listings: { entries } }, id) => entries[id];
