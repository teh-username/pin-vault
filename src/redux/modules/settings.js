import { combineReducers } from 'redux';

export const TOGGLE_PASSCODE_REQUIREMENT =
  'modules/settings/TOGGLE_PASSCODE_REQUIREMENT';
export const SET_PASSCODE = 'modules/settings/SET_PASSCODE';

/*
  Sample state:
  {
    requirePasscode:  true,
    passcodeHash: '5E884898DA28047151D0E56F8DC6292773603D0D6AABBDD62A11EF721D1542D8',
  }
 */

const initialState = {
  requirePasscode: false,
  passcodeHash: null,
};

const requirePasscode = (state = initialState.requirePasscode, action) => {
  switch (action.type) {
    case TOGGLE_PASSCODE_REQUIREMENT: {
      return !state;
    }
    default:
      return state;
  }
};

const passcodeHash = (state = initialState.passcodeHash, action) => {
  switch (action.type) {
    case SET_PASSCODE:
      return action.passcodeHash;
    default:
      return state;
  }
};

export const togglePasscodeRequirement = () => ({
  type: TOGGLE_PASSCODE_REQUIREMENT,
});

export const setPasscode = passcodeHash => ({
  type: SET_PASSCODE,
  passcodeHash,
});

export default combineReducers({
  requirePasscode,
  passcodeHash,
});

export const getCurrentPasscode = ({ settings: { passcodeHash } }) =>
  passcodeHash;

export const getPasscodeRequirement = ({ settings: { requirePasscode } }) =>
  requirePasscode;
