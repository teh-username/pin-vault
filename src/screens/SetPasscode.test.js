import React from 'react';

import {
  SetPasscode,
  mapDispatchToProps as dispatchProps,
} from './SetPasscode';

import { SET_PASSCODE } from '../redux/modules/settings';

describe('SetPasscode screen test', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<SetPasscode />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapDispatchToProps test', () => {
    it('should set the hashed passcode on submit and go back one route', () => {
      const dispatchMock = jest.fn();
      const ownPropsMock = {
        navigation: {
          goBack: jest.fn(),
        },
      };
      const props = dispatchProps(dispatchMock, ownPropsMock);
      props.handlePasscodeSubmit('password');
      expect(dispatchMock.mock.calls.length).toBe(1);
      expect(dispatchMock.mock.calls[0]).toEqual([
        {
          passcodeHash:
            '5e884898da28047151d0e56f8dc6292773603d0d6aabbdd62a11ef721d1542d8',
          type: SET_PASSCODE,
        },
      ]);

      expect(ownPropsMock.navigation.goBack.mock.calls.length).toBe(1);
    });
  });
});
