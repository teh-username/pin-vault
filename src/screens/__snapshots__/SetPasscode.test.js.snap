// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetPasscode screen test should render as expected 1`] = `<PasscodeForm />`;
